src/Aspire.Hosting.GitHub.Models/GitHubModelsExtensions.cs

```diff
@@ -85,8 +85,11 @@ public static IResourceBuilder<GitHubModelResource> WithApiKey(this IResourceBui
             throw new ArgumentException("The API key parameter must be marked as secret. Use AddParameter with secret: true when creating the parameter.", nameof(apiKey));
         }
 
-        // Remove the existing API key parameter
-        builder.ApplicationBuilder.Resources.Remove(builder.Resource.Key);
+        // Remove the existing parameter if it's the default one
+        if (builder.Resource.Key.Name == $"{builder.Resource.Name}-gh-apikey")
```

Comment:
Named based, so there is still a risk. Is it worth adding a backing field for this super niche case?